homeassistant/components/fmi/__init__.py

```diff
@@ -0,0 +1,144 @@
+"""The FMI (Finnish Meteorological Institute) component."""
+
+from datetime import date, datetime
+
+from async_timeout import timeout
+from dateutil import tz
+import fmi_weather_client as fmi
+from fmi_weather_client.errors import ClientError, ServerError
+
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_OFFSET,
+    SUN_EVENT_SUNSET,
+)
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.sun import get_astral_event_date
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    _LOGGER,
+    COORDINATOR,
+    DOMAIN,
+    FMI_WEATHER_SYMBOL_MAP,
+    MIN_TIME_BETWEEN_UPDATES,
+    UNDO_UPDATE_LISTENER,
+)
+
+PLATFORMS = ["weather"]
+
+
+def base_unique_id(latitude, longitude):
+    """Return unique id for entries in configuration."""
+    return f"{latitude}_{longitude}"
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up configured FMI."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass, config_entry) -> bool:
+    """Set up FMI as config entry."""
+    latitude = config_entry.data[CONF_LATITUDE]
+    longitude = config_entry.data[CONF_LONGITUDE]
+    time_step = config_entry.options.get(CONF_OFFSET, False)
+
+    _LOGGER.debug("Using lat: %s and long: %s", latitude, longitude)
+
+    websession = async_get_clientsession(hass)
```

Comment:
The `FMIDataUpdateCoordinator` does have `hass` so it can take care of getting the web session in its init (as the websession isn't used in this scope).